lib/udf_findhotel_utils.rb

```diff
@@ -184,6 +184,32 @@ def get_label_value(label, *keys):
           tests:       [
                        ]
       },
+      {
+          type:        :function,
+          name:        :make_im_click_batch_id,
+          description: "Returns a unique identifier for a batch of clicks reported by Intent Media.",
+          params:      "custom_dta varchar(max), campaign_tracking_code varchar(max), ad_group_tracking_code varchar(max), device varchar(max)",
+          return_type: "varchar(max)",
+          body:        %~
+            import hashlib
+            import json
+
+            key = {
+                "custom_dta": custom_dta,
+                "campaign_tracking_code": campaign_tracking_code.lower(),
+                "ad_group_tracking_code": ad_group_tracking_code.lower(),
+                "device": device.lower()}
```

Comment:
From the `label` of the IntentMedia sessions and the data available in the performance report that IM provides, those are the only 4 dimensions I can use  to generate the `click_batch_id`. Though  I am not sure it is enough to uniquely identify an ad, how could I double check that ? Who is the marketer in charge of IM ?

Reply:
@acharp yes, you can ask the marketers if there are more parameters available that we can use.